apps/src/lib/client/rpc.rs

```diff
@@ -165,6 +166,78 @@ pub async fn query_balance(ctx: Context, args: args::QueryBalance) {
     }
 }
 
+/// Query Proposals
+pub async fn query_proposal(_ctx: Context, args: args::QueryProposal) {
+    async fn print_proposal(client: &HttpClient, id: u64, details: bool) {
```

Comment:
It would be better to avoid further queries when one of the required fields is missing (especially because we're using the loop from 0 when proposal ID is not specified). One way would be to:
```suggestion
    async fn print_proposal(client: &HttpClient, id: u64, details: bool) -> Option<()> {
```
then you can use `?` on the queries, e.g.: `let author_key = gov_storage::get_author_key(id)?;` and return `Some(())` at the end to make it typecheck.
The caller should then handle `None` result to print the error message `eprintln!("No valid proposal was found with id {}", id)`. I'm not sure if we'll be deleting any data from tallied and applied proposals, but if so we might want to skip printing the error message in the loop

Reply:
At the moment, query functions like `gov_storage::get_author_key(id)` do panic in case of an invalid `id`. They do so because we use them also in the Governance VP. Can this work or do you think we should change their behavior to, say, return an `Option`?

Reply:
ah, sorry, my mistake, what I meant is to say is to use the `?` for all the queries, not the storage key, e.g. `let author = query_storage_value::<Address>(client, &author_key).await?;`, so that if one of them fails, it will return early